fedbiomed/node/secagg/_secagg_setups.py

```diff
@@ -238,11 +239,6 @@ def __init__(
         self._element = SecaggElementTypes.SERVER_KEY
         self._secagg_manager = SKManager
 
-        if not self._experiment_id or not isinstance(self._experiment_id, str):
```

Comment:
I suspect removing this test introduces a security hole (or room for a potential security hole). We need to ensure that a secagg context is attached to one experiment and cannot be used by another experiment. Thus we need to ensure the experiment id is not the empty string (which would make guessing really easy ...).

Currently in `fedbiomed.common.message` we have

```
class SecaggRequest(RequestReply, RequiresProtocolVersion):
...
      experiment_id: Optional[str]
```

So we have no guarantee about the type and value of the data delivered to `SecaggServkeySetup.__init__()`

What we could do now to simplify the code would be:

1. change `fedbiomed.common.message` to

    ```
    class SecaggRequest(RequestReply, RequiresProtocolVersion):
    ...
      experiment_id: str
    ```
3. then replace this test with a test in the base class constructor `fedbiomed.node.secagg.SecaggBaseSetup.__init__()` as we now know `experiment_id` is a `str`, and we just need to check it is non null

    ```
    if not self._experiment_id:
            errmess = f'{ErrorNumbers.FB318.value}: bad parameter `experiment_id` must be a non empty string'
            logger.error(errmess)
            raise FedbiomedSecaggError(errmess)
    ```

4. remove the equivalent test in `SecaggDHSetup.__init__()` as it is replaced by the simplified test in the base class in step 3.

Reply:
As discussed: changing `experiment_id` on `SecaggRequest` message is enough.
No need to ensure uniqueness (and secretness) of `experiment_id` currently. There might be need to partition data in the future when multiple researchers will be implemented, but the way of doing that will be decided later (and probably more scure/general mechanism)